lisa/sut_orchestrator/baremetal/bootconfig.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+from dataclasses import dataclass, field
+from pathlib import PurePosixPath
+from typing import Optional, Type
+
+from dataclasses_json import dataclass_json
+
+from lisa import schema
+from lisa.node import quick_connect
+from lisa.tools import Cat, Sed
+from lisa.util import (
+    InitializableMixin,
+    LisaException,
+    field_metadata,
+    find_group_in_lines,
+    subclasses,
+)
+from lisa.util.logger import get_logger
+
+from .schema import BootConfigSchema
+
+BOOT_LABEL = "label lisa baremetal"
+
+
+class BootConfig(subclasses.BaseClassWithRunbookMixin, InitializableMixin):
+    def __init__(
+        self,
+        runbook: BootConfigSchema,
+    ) -> None:
+        super().__init__(runbook=runbook)
+        self.boot_config_runbook: BootConfigSchema = self.runbook
+        self._log = get_logger("boot_config", self.__class__.__name__)
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return BootConfigSchema
+
+    def config(self) -> None:
+        raise NotImplementedError()
+
+
+@dataclass_json()
+@dataclass
+class PxeBootSchema(BootConfigSchema):
+    connection: Optional[schema.RemoteNode] = field(
+        default=None, metadata=field_metadata(required=True)
+    )
+    host_name: str = field(default="", metadata=field_metadata(required=True))
+    image_source: str = field(default="", metadata=field_metadata(required=True))
+
+
+class PxeBoot(BootConfig):
+    def __init__(
+        self,
+        runbook: PxeBootSchema,
+    ) -> None:
+        super().__init__(runbook=runbook)
+        self.pxe_runbook: PxeBootSchema = self.runbook
+        self._log = get_logger("pxe_boot", self.__class__.__name__)
+        self._boot_dir = "/var/lib/tftpboot/"
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "pxe_boot"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return PxeBootSchema
+
+    def config(self) -> None:
+        try:
+            self._connect_to_dhcp_server()
+        except Exception as ex:
+            self._log.exception(
+                f"failed to connect {self.pxe_runbook.connection}",
+                exc_info=ex,
+            )
+
+        node_name = f"{self.pxe_runbook.host_name}"
+        pxe_boot_config_path = self._get_pxe_config_path(node_name)
```

Comment:
It needs to consider the file not exist in a new environment.

Reply:
_get_pxe_config_path() is sure to return a valid file, else an exception is raised in its implementation if empty file is returned, so I removed the file check here.

Reply:
If the file doesn't exist, it needs to create a new file.

Reply:
The change is done, thanks